packages/app/src/cli/utilities/developer-platform-client/app-management-client.ts

```diff
@@ -720,34 +723,37 @@ const MAGIC_REDIRECT_URL = 'https://shopify.dev/apps/default-app-home/api/auth'
 
 function createAppVars(name: string, isLaunchable = true, scopesArray?: string[]): CreateAppMutationVariables {
   return {
-    appModules: [
-      {
-        uid: 'app_home',
-        specificationIdentifier: 'app_home',
-        config: JSON.stringify({
-          app_url: isLaunchable ? 'https://example.com' : MAGIC_URL,
-          embedded: isLaunchable,
-        }),
-      },
-      {
-        uid: 'branding',
-        specificationIdentifier: 'branding',
-        config: JSON.stringify({name}),
-      },
-      {
-        uid: 'webhooks',
-        specificationIdentifier: 'webhooks',
-        config: JSON.stringify({api_version: '2024-01'}),
-      },
-      {
-        uid: 'app_access',
-        specificationIdentifier: 'app_access',
-        config: JSON.stringify({
-          redirect_url_allowlist: isLaunchable ? ['https://example.com/api/auth'] : [MAGIC_REDIRECT_URL],
-          ...(scopesArray && {scopes: scopesArray.map((scope) => scope.trim()).join(',')}),
-        }),
-      },
-    ],
+    appSource: {
+      modules: [
+        {
+          uid: randomUUID(),
```

Comment:
Doesn't this fail the second time you deploy, because the UUID is different so it looks like you have 2 app homes? Maybe better to have a hardcoded UUID.